fix: issue with ComboBox initial value when used in LitTemplate

hdamr opened the pull request:
## Description

This PR fixes the issue with ComboBox when used with LitTemplate. The original issue raised when some specific order of methods were called on ComboBox element (`setReadonly`, `setPlaceholder` ...). After analysing it we realised that some calls to `getElement().setProperty(PROP_INPUT_ELEMENT_VALUE, value)` are out of sync; Which change the component `_inputElementValue` from 1 to `""`.

We fix this issue by setting `_inputElementValue` explicitly when the `value` of input is set to mitigate this randomized behaviour.  

Fixes # (issue)

## Type of change

- [X] Bugfix
- [ ] Feature

## Checklist

- [X] I have read the contribution guide: https://vaadin.com/docs-beta/latest/guide/contributing/overview/
- [X] I have added a description following the guideline.
- [X] The issue is created in the corresponding repository and I have referenced it.
- [X] I have added tests to ensure my change is effective and works as intended.
- [X] New and existing tests are passing locally with my change.
- [X] I have performed self-review and corrected misspellings.

#### Additional for `Feature` type of change

- [ ] Enhancement / new feature was discussed in a corresponding GitHub issue and Acceptance Criteria were created.

sissbruecker commented on vaadin-combo-box-flow-parent/vaadin-combo-box-flow-integration-tests/frontend/src/combobox-template-initial-value-serverside.ts:
```diff
@@ -0,0 +1,25 @@
+import {customElement, html, TemplateResult, LitElement} from "lit-element";
+
+@customElement("combobox-initial-value")
+export class TestForm extends LitElement {
```
Nit: class name does not match tag name

hdamr replied:
Done.

sissbruecker commented on vaadin-combo-box-flow-parent/vaadin-combo-box-flow-integration-tests/src/main/java/com/vaadin/flow/component/combobox/test/template/ComboBoxTemplateInitialValue.java:
```diff
@@ -0,0 +1,71 @@
+    @Id("combo")
+    private ComboBox<Bean> comboBox;
+
+    List<Bean> beans = new ArrayList<>();
```
New Sonarqube issue: CRITICAL ComboBoxTemplateInitialValue.java#L21: Make "beans" private or transient. rule

hdamr replied:
Done.

sissbruecker replied:
This is not resolved, the Sonarqube issue refers to the `beans` field that is marked in the code above.

In general please always take a look at Sonarqube issue to check if your new code introduced them.

sissbruecker commented:
Did some manual testing and could not find any issues with the change.

hdamr commented:
> Did some manual testing and could not find any issues with the change.

I reverted `NpmPackage` versions to `alpha5` and added proposed change to see if the tests fail on the CI, you can have a look [here](https://github.com/vaadin/flow-components/pull/2145)

sissbruecker commented on vaadin-combo-box-flow-parent/vaadin-combo-box-flow-integration-tests/src/main/java/com/vaadin/flow/component/combobox/test/template/ComboBoxLitWrapperPage.java:
```diff
@@ -0,0 +1,25 @@
+@Tag("combo-box-lit-page")
+@JsModule("./src/combobox-lit-page.ts")
+@Route("vaadin-combo-box/combo-box-lit-wrapper-page")
+public class ComboBoxLitWrapperPage extends LitTemplate {
```
Naming could be improved in all files/classes related to this test case to indicate that the test case is about:
- using a combobox
- in a Lit component
- that is used in a bound Lit template (not sure what the correct term is, by bound I mean it's using the template specified with `@Tag`, binding elements to properties with `@ID`)

Given that the test cases are very specific, we also can consider adding comments (to both test cases) that indicate that these are regression tests for specific Github tickets. If one of these tests creates trouble in the future, then a developer can take a look at the ticket for context.
